installation/6.docker.md

```diff
@@ -0,0 +1,205 @@
+# Docker
+In this wiki page we explain how to deploy MAD using docker.
+
+
+If you do not have a clue about docker, you maybe want to check out:
+ - https://www.docker.com/why-docker
+ - https://www.docker.com/resources/what-container
+
+## Get Docker
+First of all, you have to install Docker CE and docker-compose on your system.
+
+- Docker CE: just execute that script https://get.docker.com/ or read through https://docs.docker.com/install/ 
+- Docker-compose: https://docs.docker.com/compose/install/
+
+These sites are well documented and if you follow the install instructions, you are good to go.
+
+
+## 1 Setup MAD and Rocketmap database.
+In this section we explain how to setup MAD and a Rocketmap database using docker-compose.
+
+### 1.1 Preparations:
+
+You can just copy & paste this to do what is written below: 
+```
+mkdir MAD-docker && \
+cd MAD-docker && \
+mkdir mad && \
+mkdir mad/configs && \
+touch docker-compose.yml && \
+mkdir docker-entrypoint-initdb && \
+wget -O docker-entrypoint-initdb/rocketmap.sql https://raw.githubusercontent.com/Map-A-Droid/MAD/dev/SQL/rocketmap.sql && \
+cd mad/configs/ && \
+wget -O config.ini https://raw.githubusercontent.com/Map-A-Droid/MAD/master/configs/config.ini.example && \
+wget -O mappings.json https://raw.githubusercontent.com/Map-A-Droid/MAD/master/configs/mappings_example.json  && \
+mkdir geofences && cd ../../
+```
+This will: 
+1. Create a directory `MAD-docker`. 
+2. Create a file `docker-compose.yml`.
+3. Create a directory `MAD-docker/mad`. (here we store MAD related stuff)
+4. Create a directory `MAD-docker/mad/configs`. (here we store config files for MAD).
+Here you store your `config.ini`, `mappings.json` and a directory `geofences` (which contains your geofences).
+Examples for these files can be found @github https://github.com/Map-A-Droid/MAD/tree/master/configs
+5. Download the Rocketmap Database Schema: https://gist.github.com/sn0opy/fb654915180cfbd07d5a30407c286995 and store it
+as a file `rocketmap-osm-cec.sql`
+
+Your directory should now look like this:
+ ``` 
+ --MAD-docker/
+  |-- docker-compose.yml
+  |-- rocketmap-osm-cec.sql
+  |-- mad/
+    |-- configs/
+      |-- config.ini
+      |-- mappings_.json
+      |-- geofences/
+ ```
+ 
+### 1.2 Writing the docker-compose file.
+We use `docker-compose` to deploy and manage our services.
+
+Fill `docker-compose.yml` with the following content. Below we explain the details of every service.
+```yaml
+version: '2.4'
+services:
+  mad:
+    container_name: pokemon_mad
+    image: mapadroid/map-a-droid
+    restart: always
+    volumes:
+      - /etc/timezone:/etc/timezone:ro
+      - /etc/localtime:/etc/localtime:ro
+      - ./mad/configs/geofences:/usr/src/app/configs/geofences
+      - ./mad/configs/config.ini:/usr/src/app/configs/config.ini
+      - ./mad/configs/mappings.json:/usr/src/app/configs/mappings.json
+      - ./volumes/mad/version.json:/usr/src/app/version.json
```

Comment:
I went through the entire document on a clean system and this is the only line I ran into an issue with. Docker is attempting to link this as a folder instead of a file which causes MAD to fail. On my production system I'm not mounting this file and don't think it's super important to keep this line. MAD will recreate it on a new system so I'm not sure I see the value.

Reply:
I agree, it is obsolete to mount the `version.json`. 
324ea5a fixes this, and your other recommendations.